ashley/src/main/kotlin/ktx/ashley/entities.kt

```diff
@@ -65,3 +66,15 @@ operator fun <T : Component> Entity.contains(mapper: ComponentMapper<T>): Boolea
  * @see Entity.remove
  */
 inline fun <reified T : Component> Entity.remove(): Component? = remove(T::class.java) as? T
+
+/**
+ * Adds a [Component] to this [Entity]. If a [Component] of the same type already exists, it'll be replaced.
+ *
+ * @return the Entity for easy chaining
+ * @see Entity.add
+ */
+inline fun <reified T : Component> Entity.add(engine: Engine, configure: (@AshleyDsl T).() -> Unit = {}): Entity {
```

Comment:
Why is `Entity` returned by this method? I understand that Ashley does this for chaining, but in their case you already have access to the `Component` instance. In this case we're creating a new `Component`, so it makes sense to return it. I think the return type should be `T`.

Consider renaming this to `addComponent`. If the reified type is extracted e.g. from variable, the method name becomes confusing as it reads as "add engine". For example:

```kotlin
val transform: Transform = entity.add(engine)
```

Reply:
I thought it makes sense to have the same behavior as the original ashley function and returning the entity does not really help a lot because you anyway can specify component stuff in the lambda?

currently you could do something like:
```kotlin
entity.add<Damaged> {
  damage = 3f
}.add<Knockback> {
  force = 2.5f
}
```

When you return the component, you can no longer chain except you use kotlin's apply/run/with/let.

What do you think? Shall I still change it?

Reply:
I don't find chaining particularly cleaner than:

```kotlin
entity.apply {
  add<Damaged> {
    damage = 3f
  }
  add<Knockback> {
    force = 2.5f
  }
}
```

I think chaining methods are usually a legacy from Java that make up for the lack of DSL capabilities.

Consider that if you need a reference to the created component, you'd have to do this:
```kotlin
lateinit var damaged: Damaged
entity.add<Damaged> {
  damage = 3f
  damaged = this
}
```

Or this:
```kotlin
entity.add<Damaged> {
  damage = 3f
}
val damaged = entity.get<Damaged>()
```

Instead of this:
```kotlin
val damaged = entity.add<Damaged> {
  damage = 3f
}
```

As a rule of thumb - if a utility method creates something, we want a reference to it. We already have a reference to the entity, but we might need a reference to the component we have just created.